src/hooks/useAuthRequired.tsx

```diff
@@ -0,0 +1,100 @@
+/*!
+ * Copyright (c) 2017-Present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import * as React from 'react';
+import { OnAuthRequiredFunction, IOktaContext } from '../types';
+import { getRelativeUri } from '../utils';
+
+export interface AuthRequiredOptions {
+  onAuthRequired?: OnAuthRequiredFunction;
+  requiresAuth?: boolean; // default: true
+}
+
+export interface AuthRequiredHook {
+  isAuthenticated: boolean;
+  loginError: Error | null;
+}
+
+const useAuthRequired = (
```

Comment:
I understand the motivation for moving this (mostly) common logic into a single hook, however I think it's best to leave this logic in the `SecureRoute` implementations respectively

Reply:
Do you mean copy the logic inside `useAuthRequired` hook to `SecureRoute` and `AuthRequired`? 
And remove `useAuthRequired` hook?